osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/virtualizationconnector/VCDeleteMetaTask.java

```diff
@@ -57,20 +60,27 @@ public void executeTransaction(EntityManager em) throws Exception {
             OSCEntityManager<VirtualizationConnector> vcEntityMgr = new OSCEntityManager<>(VirtualizationConnector.class, em, this.txBroadcastUtil);
             VirtualizationConnector connector = vcEntityMgr.findByPrimaryKey(this.vc.getId());
 
+            cleanKeystoneProviderConnection(connector);
+
             SslCertificateAttrEntityMgr sslCertificateAttrEntityMgr = new SslCertificateAttrEntityMgr(em, this.txBroadcastUtil);
             sslCertificateAttrEntityMgr.removeCertificateList(connector.getSslCertificateAttrSet());
             vcEntityMgr.delete(this.vc.getId());
         } catch (Exception ex) {
             throw ex;
-        } finally{
-        	 // Unlock VC.
+        } finally {
+            // Unlock VC.
             if (vcUnlockTask != null) {
                 log.info("Releasing lock for VC '" + this.vc.getName() + "'");
                 LockManager.getLockManager().releaseLock(new LockRequest(vcUnlockTask));
             }
         }
     }
 
+    private void cleanKeystoneProviderConnection(VirtualizationConnector vc) throws EncryptionException {
```

Comment:
how does this work for jobs? is the keystone provider used for all api calls?

How does this get cleaned up in those instances?